plugins/com.google.cloud.tools.eclipse.swtbot/src/com/google/cloud/tools/eclipse/swtbot/SwtBotTreeUtilities.java

```diff
@@ -364,19 +364,27 @@ public boolean test() throws Exception {
    * 
    * @throws TimeoutException if the child does not appear within the default timeout
    */
-  public static void waitUntilTreeItemhasChild(SWTWorkbenchBot bot, final SWTBotTreeItem treeItem,
+  public static void waitUntilTreeItemHasChild(SWTWorkbenchBot bot, final SWTBotTreeItem treeItem,
       final String childText) {
-    bot.waitUntil(new DefaultCondition() {
-      @Override
-      public String getFailureMessage() {
-        return "Tree items never disappeared";
-      }
-
-      @Override
-      public boolean test() throws Exception {
-        return treeItem.getNodes().contains(childText);
+    try {
+      bot.waitUntil(new DefaultCondition() {
+        @Override
+        public String getFailureMessage() {
+          return "Tree item never appeared";
+        }
+  
+        @Override
+        public boolean test() throws Exception {
+          return treeItem.getNodes().contains(childText);
+        }
+      });
+    } catch (TimeoutException e) {
+      System.err.println(treeItem + ": expanded? " + treeItem.isExpanded());
```

Comment:
Would this be better in `getFailureMessage()`?

Reply:
Good idea.